[SYCL][TEST] Remove check_has.cpp dependency on metadata order

jinge90 opened the pull request:
Previously, we committed https://github.com/intel/llvm/pull/9642 aiming to break dependency on metadata order for check_has.cpp but the commit was broken unexpectedly. This PR re-lands it.

jinge90 commented:
Hi, @steffenlarsen and @maarquitos14 
Previously, we added declarations for "hypotf" when oneAPI compiler is used on Windows platform but the change lead to device_has.cpp lit test failure due to meta data order is changed (aspects and srcloc), so we need to fix the test.
Previous test fix is: https://github.com/intel/llvm/pull/9642

maarquitos14 commented:
> Hi, @steffenlarsen and @maarquitos14 Previously, we added declarations for "hypotf" when oneAPI compiler is used on Windows platform but the change lead to device_has.cpp lit test failure due to meta data order is changed (aspects and srcloc), so we need to fix the test. Previous test fix is: #9642

@jinge90 why was it reverted earlier? Did it cause any issue?

jinge90 commented:
> > Hi, @steffenlarsen and @maarquitos14 Previously, we added declarations for "hypotf" when oneAPI compiler is used on Windows platform but the change lead to device_has.cpp lit test failure due to meta data order is changed (aspects and srcloc), so we need to fix the test. Previous test fix is: #9642
> 
> @jinge90 why was it reverted earlier? Did it cause any issue?

Hi, @maarquitos14 
It was not reverted but https://github.com/intel/llvm/pull/10317 changed the test and removed the 2-time checks, our previous test didn't cause any issue. Maybe the author thought that 2-time checks seemed to be unnecessary.
Thanks very much.

maarquitos14 commented:
> > > Hi, @steffenlarsen and @maarquitos14 Previously, we added declarations for "hypotf" when oneAPI compiler is used on Windows platform but the change lead to device_has.cpp lit test failure due to meta data order is changed (aspects and srcloc), so we need to fix the test. Previous test fix is: #9642
> > 
> > 
> > @jinge90 why was it reverted earlier? Did it cause any issue?
> 
> Hi, @maarquitos14 It was not reverted but #10317 changed the test and removed the 2-time checks, our previous test didn't cause any issue. Maybe the author thought that 2-time checks seemed to be unnecessary. Thanks very much.

I see, thanks for explaining.

maarquitos14 reviewed (commented):
LGTM.

jinge90 commented:
Hi, @intel/llvm-gatekeepers 
Could you help merge this patch?
Thanks very much.
